src/main/java/io/vertx/core/http/impl/pool/Pool.java

```diff
@@ -195,6 +200,27 @@ private boolean acquireConnection(Waiter<C> waiter) {
     }
   }
 
+  /**
+   * Close all unused connections with a {@code timestamp} greater than expiration timestamp.
+   *
+   * @param timestamp the timestamp value
+   * @return the number of closed connections when calling this method
+   */
+  public synchronized int closeIdle(long timestamp) {
+    int removed = 0;
+    if (waitersQueue.isEmpty() && available.size() > 0) {
+      List<Holder<C>> copy = new ArrayList<>(available);
```

Comment:
Why do you need to work on a copy here?

Reply:
because `closeConnection(conn.connection)` will modify the `available` , I could have used an `Iterator` to remove the elements but it means to update other state (like the weight) and therefore I prefer to copy this code (that is clearly not an hotspot) and avoid duplicate code that modifies the pool state.